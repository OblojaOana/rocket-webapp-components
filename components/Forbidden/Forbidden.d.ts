export interface ForbiddenProps {

    /**
     * Text to be displayed
     * @default "Not allowed to see this page!"
     */
    forbiddenText?: string;

    /**
     * Demos:
     * 
     * - https://bit.dev/totalsoft_oss/react-mui/forbidden
     * 
     *  Take a look over the code for further information:https://github.com/osstotalsoft/rocket-webapp-components/blob/develop/components/Forbidden/Forbidden.js
     */
}

export default function Forbidden(props: ForbiddenProps): JSX.Element;