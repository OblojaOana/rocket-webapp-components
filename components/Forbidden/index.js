import Forbidden from "../../core/Forbidden";
export default Forbidden;
