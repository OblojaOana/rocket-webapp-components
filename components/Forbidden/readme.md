##### Copy this snapshot in the overview panel

```js
import React from 'react';
import Forbidden from '@bit/totalsoft_oss.react-mui.forbidden';

export default (
	<Forbidden/>
));
```