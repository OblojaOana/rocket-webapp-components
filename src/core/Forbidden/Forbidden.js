import React from 'react';
import { Card, makeStyles } from '@material-ui/core';
import { useTranslation } from 'react-i18next';
import Typography from '@material-ui/core/Typography';
import forbiddenStyle from './forbiddenStyle';

const useStyles = makeStyles(forbiddenStyle);

const Forbidden = () => {
    const { t } = useTranslation();
    const classes = useStyles();

    return (
        <Card className={classes.cardStyle}>
            <Typography align={"center"} variant={"h6"} className={classes.textStyle}>{t('Forbidden')}</Typography>
        </Card>
    )
}

export default Forbidden;