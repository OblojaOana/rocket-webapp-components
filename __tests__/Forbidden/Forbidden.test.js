import React from "react";
import renderer from "react-test-renderer";
import Forbidden from "../../components/Forbidden/Forbidden";

describe("Render Forbidden", () => {
  it("render correctly Forbidden component", () => {
    const component = renderer.create(<Forbidden />);

    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
