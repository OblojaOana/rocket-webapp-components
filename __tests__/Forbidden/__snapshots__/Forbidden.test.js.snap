// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forbidden Snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-cardStyle-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <h6
    className="MuiTypography-root makeStyles-textStyle-2 MuiTypography-h6 MuiTypography-alignCenter"
  >
    Not allowed to see this page!
  </h6>
</div>
`;
